src/main/common/streambuf.h

```diff
@@ -18,15 +18,19 @@
 #pragma once
 
 #include <stdint.h>
+#include <stdbool.h>
 
 // simple buffer-based serializer/deserializer without implicit size check
 // little-endian encoding implemneted now
 
 typedef struct sbuf_s {
-    uint8_t *ptr;          // data pointer must be first (sbuff_t* is equivalent to uint8_t **)
-    uint8_t *end;
+    uint8_t *bufPtr;          // data pointer must be first (sbuff_t* is equivalent to uint8_t **)
```

Comment:
I don't like these name changes. `ptr` and `end` are clear in meaning for the sbuf. `bufPtr` is just redundant.

Reply:
That was intentional - to make sure old code that was doing initialization manually will throw an error when compiling.